dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/message/NettyResponseMessage.java

```diff
@@ -0,0 +1,706 @@
+
+        if (HttpMethod.HEAD.toString().equals(getServiceContext().getRequest().getMethod())) {
+            return false;
+        }
+
+        boolean bodyExpected = super.isBodyExpected();
+
+        if (!bodyExpected) {
+            bodyExpected = containsHeader(HttpHeaderNames.CONTENT_ENCODING.array()) || containsHeader(HttpHeaderNames.CONTENT_RANGE.array());
+        }
+
+        return bodyExpected && isBodyAllowedForStatusCode();
+
+    }
+
+    @Override
+    public boolean isBodyAllowed() {
+        if (super.isBodyAllowed()) {
+
+            if (HttpMethod.HEAD.toString().equals(getServiceContext().getRequest().getMethod())) {
+                return false;
+            }
+            return isBodyAllowedForStatusCode();
+        }
+        return false;
+    }
+
+    @Override
+    public void setConnection(ConnectionValues value) {
+        if (value.getName().equalsIgnoreCase(HttpHeaderValues.CLOSE.toString()))
+            nettyResponse.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        else if (value.getName().equalsIgnoreCase(HttpHeaderValues.KEEP_ALIVE.toString()))
+            nettyResponse.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.KEEP_ALIVE);
+    }
+
+    @Override
+    public void setConnection(ConnectionValues[] values) {
+        nettyResponse.headers().set(HttpHeaderNames.CONNECTION, values);
+    }
+
+    @Override
+    public ConnectionValues[] getConnection() {
+        List<String> test = nettyResponse.headers().getAll(HttpHeaderNames.CONNECTION);
+        List<ConnectionValues> values = new ArrayList<ConnectionValues>();
+        for (String header : test) {
+            values.add(ConnectionValues.match(header, 0, header.length()));
+        }
+        return (ConnectionValues[]) values.toArray();
+    }
+
+    @Override
+    public boolean isKeepAliveSet() {
+        return HttpUtil.isKeepAlive(nettyResponse);
+    }
+
+    @Override
+    public boolean isConnectionSet() {
+        return this.containsHeader(HttpHeaderKeys.HDR_CONNECTION);
+    }
+
+    @Override
+    public void setContentEncoding(ContentEncodingValues value) {
+        headers.set(HttpHeaderNames.CONTENT_ENCODING, value.getName());
+    }
+
+    @Override
+    public void setContentEncoding(ContentEncodingValues[] values) {
+        headers.remove(HttpHeaderNames.CONTENT_ENCODING);
+        for(ContentEncodingValues value : values){
+            headers.add(HttpHeaderNames.CONTENT_ENCODING, value.getName());
+        }
+    }
+
+    @Override
+    public ContentEncodingValues[] getContentEncoding() {
+        List<String> encodingHeaders = headers.getAll(HttpHeaderNames.CONTENT_ENCODING);
+        List<ContentEncodingValues> values = new ArrayList<ContentEncodingValues>();
+        for (String encoding : encodingHeaders) {
+            values.add(ContentEncodingValues.match(encoding, 0, encoding.length()));
+        }
+        return (ContentEncodingValues[]) values.toArray();
+    }
+
+    @Override
```

Comment:
We should probably remove the unimplemented methods especially if they are already in the base http message class

Reply:
I'll add to post beta clean up issue